File: Java/Excel Sheet Column Number.java
/**
 * Problem: https://leetcode.com/problems/excel-sheet-column-number/
 */
public class Solution {
    public int titleToNumber(String s) {
        int sum = 0;
        for(int i=0; i<s.length(); i++) {
            sum = 26 * sum + s.charAt(i) - 'A' + 1;
        }
        return sum;
    }
}

File: Swift/excel-sheet-column-number.swift
/**
 * https://leetcode.com/problems/excel-sheet-column-number/
 * 
 * 
 */ 
// Date: Mon Aug 10 10:16:08 PDT 2020
class Solution {
    func titleToNumber(_ s: String) -> Int {
        var result = 0
        for c in s.unicodeScalars {
            result = result * 26 + Int(c.value) - 64
        }
        return result
    }
}
/**
 * https://leetcode.com/problems/excel-sheet-column-number/
 * 
 * 
 */ 
// Date: Mon Aug 10 10:18:33 PDT 2020
class Solution {
    func titleToNumber(_ s: String) -> Int {
        return s.unicodeScalars.reduce(0) { $0 * 26 + Int($1.value) - 64 }
    }
}
